### ISSUE TITLE
Heap Overflow in tcpcapinfo.c

### ISSUE THREAD

**Issue by entropy1337** [appneta/tcpreplay#487]
pcap_ph seem to be passed to other variables unsafely.

**Comment by SegfaultMasters** [comment]
More info?

**Comment by entropy1337** [comment]
hi,
memcpy utilization tcpcapinfo.c doesn't seem to handle the data properly
and overflowing the heap with excess data.

file size   = 64984 bytes
magic       = 0xa1b2c3d4 (tcpdump) (little-endian)
version     = 2.4
thiszone    = 0x00000000
sigfigs     = 0x00000000
snaplen     = 65535
linktype    = 0x00000001
Packet OrigLen Caplen Timestamp Csum Note
1   93   92 4c33f183.71f37 fc282 OK
*** buffer overflow detected ***: ./tcpcapinfo terminated
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x777e5)[0x7fc1989f07e5]
/lib/x86_64-linux-gnu/libc.so.6(__fortify_fail+0x5c)[0x7fc198a9215c]
/lib/x86_64-linux-gnu/libc.so.6(+0x117160)[0x7fc198a90160]
/lib/x86_64-linux-gnu/libc.so.6(+0x1175ca)[0x7fc198a905ca]
./tcpcapinfo[0x402b50]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0)[0x7fc198999830]
./tcpcapinfo[0x404009]
======= Memory map: ========
00400000-00476000 r-xp 00000000 08:02 6696309
/home/targets/tcpreplay/tcpreplay-4.2.5/src/tcpcapinfo
00675000-00676000 r--p 00075000 08:02 6696309
/home/targets/tcpreplay/tcpreplay-4.2.5/src/tcpcapinfo
00676000-00677000 rw-p 00076000 08:02 6696309
/home/targets/tcpreplay/tcpreplay-4.2.5/src/tcpcapinfo
00677000-00678000 rw-p 00000000 00:00 0
025e7000-02608000 rw-p 00000000 00:00 0
[heap]

**Comment by fklassen** [comment]
Need more info to reproduce. What version are you using (should be building off 4.3 branch)? What is the command-line argument? Please provide failing PCAP file.

**Comment by entropy1337** [comment]
Hello Fred,
I am using the one from main page, version tcpreplay-4.2.5.
should I share the link here on github or email you the link ?
my email is 3ntr0py1337@gmail.com in case you need it.

**Comment by fklassen** [comment]
Received direct email with instructions on how to recreate.

First I compiled with aft-clang (latest version). Ran `afl-fuzz` for a while with both 4.2.5 and 4.3-beta2 build, but didn't see crash. Apparently it is because I don't have a 50MB memory limit set up. 

However I was able to recreate a crash on 4.2.5 using AFL output sent to me.

Also I ran afl-fuzz for a while and didn’t see crashes. I don’t quite understand the instructions
regarding 50MB restriction...

```
fklassen@debian-fkk-stretch:/tmp/crash/crashes$ ./tcpcapinfo-4.2.5 id:000000,sig:06,src:000000,op:flip1,pos:32
file size   = 64984 bytes
magic       = 0xa1b2c3d4 (tcpdump) (little-endian)
version     = 2.4
thiszone    = 0x00000000
sigfigs     = 0x00000000
snaplen     = 65535
linktype    = 0x00000001
Packet	OrigLen		Caplen		Timestamp	Csum	Note
1	  93		  92		4c33f183.71f37	fc282	OK
2	16896		16896		33f18392.722094cFile truncated!  Unable to jump to next packet.
```

With version 4.3-beta2 I don’t see a crash, probably because of https://github.com/appneta/tcpreplay/blob/4.3/src/common/utils.c#L151
Implemented with PR #484.

```
fklassen@debian-fkk-stretch:/tmp/crash/crashes$ ./tcpcapinfo-4.3-beta2 id:000000,sig:06,src:000000,op:flip1,pos:32
file size   = 64984 bytes
magic       = 0xa1b2c3d4 (tcpdump) (little-endian)
version     = 2.4
thiszone    = 0x00000000
sigfigs     = 0x00000000
snaplen     = 65535
linktype    = 0x00000001
Packet	OrigLen		Caplen		Timestamp	Csum	Note
1	  93		  92		4c33f183.71f37	fc282	OK
2	16896		16896		33f18392.722094c	f11ca63	BAD_TS
3	1948269172		1885692771		2e362e33.410a0d36

Capture file appears to be damaged or corrupt.
Contains packet of size 1885692771, bigger than snap length 65535
```

I’ll close the issue for now. If anyone is able to recreate it from the 4.3 branch, please reopen.
